src/moq_instance.cpp

```diff
@@ -0,0 +1,1687 @@
+                             MoQTerminationReason::PROTOCOL_VIOLATION,
+                             "Control bidir stream not created");
+            return;
+        }
+
+        _transport->enqueue(conn_ctx.conn_id,
+                            *conn_ctx.ctrl_data_ctx_id,
+                            std::move(data),
+                            { MethodTraceItem{} },
+                            0,
+                            2000,
+                            0,
+                            { true, false, false, false });
+
+    }
+
+    void MoQInstance::send_client_setup()
+    {
+        StreamBuffer<uint8_t> buffer;
+        auto client_setup = MoqClientSetup{};
+
+        client_setup.num_versions = 1;      // NOTE: Not used for encode, verison vector size is used
+        client_setup.supported_versions = { MOQT_VERSION };
+        client_setup.role_parameter.param_type = static_cast<uint64_t>(ParameterType::Role);
+        client_setup.role_parameter.param_length = 0x1; // NOTE: not used for encode, size of value is used
+        client_setup.role_parameter.param_value = { 0x03 };
+        client_setup.endpoint_id_parameter.param_value.assign(_client_config.endpoint_id.begin(),
+                                                              _client_config.endpoint_id.end());
+
+        buffer << client_setup;
+
+        auto &conn_ctx = _connections.begin()->second;
+
+        send_ctrl_msg(conn_ctx, buffer.front(buffer.size()));
+    }
+
+    void MoQInstance::send_server_setup(ConnectionContext& conn_ctx)
+    {
+        StreamBuffer<uint8_t> buffer;
+        auto server_setup = MoqServerSetup{};
+
+        server_setup.selection_version = { conn_ctx.client_version };
+        server_setup.role_parameter.param_type = static_cast<uint64_t>(ParameterType::Role);
+        server_setup.role_parameter.param_length = 0x1; // NOTE: not used for encode, size of value is used
+        server_setup.role_parameter.param_value = { 0x03 };
+        server_setup.endpoint_id_parameter.param_value.assign(_server_config.endpoint_id.begin(),
+                                                              _server_config.endpoint_id.end());
+
+
+        buffer << server_setup;
+
+        _logger->debug << "Sending SERVER_SETUP to conn_id: " << conn_ctx.conn_id << std::flush;
+
+        send_ctrl_msg(conn_ctx, buffer.front(buffer.size()));
+    }
+
+    void MoQInstance::send_announce(ConnectionContext& conn_ctx, std::span<uint8_t const> track_namespace)
+    {
+        StreamBuffer<uint8_t> buffer;
+        auto announce = MoqAnnounce{};
+
+        announce.track_namespace.assign(track_namespace.begin(), track_namespace.end());
+        announce.params = {};
+        buffer <<  announce;
+
+        std::vector<uint8_t> net_data = buffer.front(buffer.size());
+
+        _logger->debug << "Sending ANNOUNCE to conn_id: " << conn_ctx.conn_id << std::flush;
+
+        send_ctrl_msg(conn_ctx, buffer.front(buffer.size()));
```

Comment:
Net_data isn't used, should it be? Looks like it uses the same bytes.

Reply:
thanks for catching that. It was left over and should have been removed. I've removed it.